src/Native/System.Native/pal_networking.cpp

```diff
@@ -428,7 +429,168 @@ static void ConvertHostEntPlatformToPal(HostEntry& hostEntry, hostent& entry)
     }
 }
 
-#if HAVE_GETHOSTBYNAME_R
+#if !HAVE_THREAD_SAFE_GETHOSTBYNAME_AND_GETHOSTBYADDR
+#if !HAVE_GETHOSTBYNAME_R
+static int copy_hostent(struct hostent* from, struct hostent* to,
+                        char* buffer, size_t buflen)
+{
+    int i, len;
+    char* ptr = buffer;
+
+    to->h_addrtype = from->h_addrtype;
+    to->h_length = from->h_length;
+    to->h_name = static_cast<char*>(ptr);
+
+    /* copy hostname to buffer */
+    len = static_cast<int>(strlen(from->h_name)+1);
+    SafeStringCopy(ptr, buflen, from->h_name);
+    ptr += len;
+
+    if (strlen(ptr)+1 > buflen)
+    {
+        return ERANGE;
+    }
+
+    /* copy aliases to buffer */
+    to->h_aliases = reinterpret_cast<char**>(ptr);
+    for(i = 0; from->h_aliases[i]; i++);
+    ptr += (i+1) * static_cast<int>(sizeof(char*));
+
+    if (strlen(ptr)+1 > buflen)
+    {
+        return ERANGE;
+    }
+
+    for (i = 0; from->h_aliases[i]; i++)
+    {
+        len = static_cast<int>(strlen(from->h_aliases[i])+1);
+        if (len >= 0 && static_cast<size_t>((ptr-buffer)+len) < buflen)
+        {
+            to->h_aliases[i] = ptr;
+            SafeStringCopy(ptr, static_cast<int>(ptr-buffer), from->h_aliases[i]);
+            ptr += len;
+        }
+        else
+        {
+            return ERANGE;
+        }
+    }
+
+    if (strlen(to->h_aliases[i])+1 > buflen)
+    {
+        return ERANGE;
+    }
+
+    to->h_aliases[i] = NULL;
+
+    /* copy addr_list to buffer */
+    to->h_addr_list = reinterpret_cast<char**>(ptr);
+    for(i = 0; reinterpret_cast<int*>(from->h_addr_list[i]) != 0; i++);
+    ptr += (i+1) * static_cast<int>(sizeof(int*));
+
+    for (i = 0; from->h_addr_list[i] != nullptr; i++)
+    {
+        len = sizeof(int);
+        if (static_cast<size_t>((ptr-buffer)+len) < buflen)
+        {
+            to->h_addr_list[i] = ptr;
+            memcpy(ptr, from->h_addr_list[i], static_cast<size_t>(len));
+            ptr += len;
+        }
+        else
+        {
+            return ERANGE;
+        }
+    }
+
+    if (strlen(to->h_addr_list[i])+1 > buflen)
+    {
+        return ERANGE;
+    }
+
+    to->h_addr_list[i] = NULL;
+    return 0;
+}
```

Comment:
Ok thanks. IMO the preprocessor conditions are only satisfied on NetBSD for these three functions and compiler won't compile this code on other supported operating systems. Would it be acceptable to wrap these three under `#ifdef __NETBSD__` (since NetBSD work is presently 'in progress' at best) or just put a `FIXME` note explaining the situation while keeping the existing preprocessor condition?